src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/IUserTimeZoneService.cs

```diff
@@ -11,11 +11,27 @@ public interface IUserTimeZoneService
     /// Gets the time zone for the given user.
     /// </summary>
     /// <param name="user">The <see cref="IUser"/>.</param>
+    [Obsolete("This method is obsolete and will be removed in future release. Instead use GetAsync(userName).")]
     ValueTask<ITimeZone> GetAsync(IUser user);
 
     /// <summary>
     /// Updates the time zone for the given user.
     /// </summary>
     /// <param name="user">The <see cref="IUser"/>.</param>
+    [Obsolete("This method is obsolete and will be removed in future release. Instead use ForgetCacheAsync(userName).")]
     ValueTask UpdateAsync(IUser user);
+
+    /// <summary>
+    /// Gets the time zone for the given user.
+    /// </summary>
+    /// <param name="userName">The user name.</param>
+    ValueTask<ITimeZone> GetAsync(string userName)
+        => ValueTask.FromResult<ITimeZone>(null);
+
+    /// <summary>
+    /// Forget the time zone for the given user.
+    /// </summary>
+    /// <param name="userName">The user name.</param>
+    ValueTask ForgetCacheAsync(string userName)
```

Comment:
This assumes implementations will cache. It feels very weird, I don't think we have done this before.

Reply:
We have other ways to manage cache invalidation. It feels we are going backwards from the initial issue we were fixing.

Reply:
Currently the UpdateAsync(IUser user) invalidate the cache.

Reply:
@sebastienros I don't think we can tag a cache used by the distributed cache which will invalidate the cache based on the user tag. Do you have an idea in mind to auto invalidate the cache so we don't need such a method here?

Reply:
@sebastienros I actually removed the ForgetCacheAsync method from the interface. Now this interface needs a single method to get the time zone. I implemented the `IUserEventHandler` to invalidate the cache automatically.